endpoint/endpoint.go

```diff
@@ -38,3 +38,22 @@ func Chain(outer Middleware, others ...Middleware) Middleware {
 type Failer interface {
 	Failed() error
 }
+
+// EndpointNameMiddleware populates the context with a common name for the endpoint.
+// It can be used in subsequent endpoints in the chain to identify the operation.
+func EndpointNameMiddleware(name string) Middleware {
+	return func(next Endpoint) Endpoint {
+		return func(ctx context.Context, req interface{}) (interface{}, error) {
+			ctx = context.WithValue(ctx, ContextKeyEndpointName, name)
+
+			return next(ctx, req)
+		}
+	}
+}
+
+type contextKey int
+
+const (
+	// ContextKeyEndpointName is populated in the context by EndpointNameMiddleware.
+	ContextKeyEndpointName contextKey = iota
```

Comment:
So instead of exporting the context key it is much better to add a method to fetch the endpoint name. The getting and setting of context values should be a concern of the owning package, not the consuming packages

Reply:
100% agree, I tried to follow the existing pattern in go-kit. Happy to add a getter instead of the exported key.

Reply:
I added a getter for the context key, but left it exported to follow existing go-kit examples. Should I make it unexported?